src/ecs/transform/components/local_transform.rs

```diff
@@ -3,19 +3,25 @@
 use std::ops::{Deref, DerefMut};
 use std::sync::atomic::{AtomicBool, Ordering};
 
-use cgmath::{Matrix3, Matrix4, Quaternion, Vector3};
+use cgmath::{Euler, Matrix3, Matrix4, Quaternion, Vector3, Vector4};
 
 use ecs::{Component, VecStorage};
 
 /// Raw transform data.
 #[derive(Debug)]
 pub struct InnerTransform {
-    /// Translation/position vector [x, y, z]
-    pub translation: [f32; 3],
+    /// Forward vector [x, y, z]
+    pub forward: [f32; 3],
+    /// Right vector [x, y, z]
+    pub right: [f32; 3],
     /// Quaternion [w (scalar), x, y, z]
     pub rotation: [f32; 4],
     /// Scale vector [x, y, z]
     pub scale: [f32; 3],
+    /// Translation/position vector [x, y, z]
+    pub translation: [f32; 3],
+    /// Up vector [x, y, z]
+    pub up: [f32; 3],
```

Comment:
Are all these really necessary? This nearly doubles the size of the transform.

Reply:
I think you can just get these from the quaternion:
```rust
let w = transform.rotation[0];
let x = transform.rotation[1];
let y = transform.rotation[2];
let z = transform.rotation[3];
let up = [
    2 * (x * y - w * z), 
    -2 * (x * x + z * z), 
    2 * (y * z + w * x),
];
let right = [
    1 - 2 * (y * y + z * z),
    2 * (x * y + w * z),
    2 * (x * z - w * y),
];
let forward = [
    2 * (x * z + w * y),
    2 * (y * x - w * x),
    1 - 2 * (x * x + y * y),
];
```

Reply:
Which approach we go with kind of depends on how often these are accessed.  If we can it'd be nice to just use results from previous computations, so we can avoid needlessly repeating calculations, but if they're accessed infrequently it might be better to just do the calculation on demand.

Reply:
yes, for movement and rotation, I have to know how the player set up the orientation of things. Alternatively, we could define a static orientation.... Previously, these fields were only on the camera, but we want to have movement freedom on all objects, right :)

tbh, if you have a better idea on how to implement it, or if you think, that we do not need that kind of functionality, I am open for discussions. For now, I put everything in there which we talked about in the issues and here.

Reply:
You could always just add a secondary component if you are using these a lot, if we add them to the `LocalTransform` though then we are doubling the size of the component (which leads to less cache efficiency) and if the user doesn't use these then it is wasted.

Reply:
btw, your above calculations only give you the local system, not the global (as in the base orientation).

I already thought about making another component. Doing the separation might be interesting for static meshes, which don't change their rotation and position during the simulation. The only downside I can see is that the user would have to pass the matching LocalTransform to every movement or use a movement-builder (which passes the LocalTransform). I think, the downside is not too bad, so I will probably implement that.

Reply:
or... we could set the global orientation for _all_ objects as a resource. That would mean another parameter - in case you don't have a better idea on how to get the resource from within LocalTransform...

Well, thinking about this stuff, I am wondering. LocalTransform is a component. Is it even the right place for those calculations? That movement-builder seems better and better (though how good is it performance-wise?)